### DISCUSSION TITLE
Why use additional state management?

### DISCUSSION THREAD

**Discussion by awmiklovic** [react-hook-form/react-hook-form#6299] [Q&A] [answered]
The Wizard Form example recommends using state management alongside rhf. Doesn't rhf provide state management through the path of registering fields? The values can be updated and accessed as needed. What is the benefit of tracking state separately?

https://react-hook-form.com/advanced-usage#WizardFormFunnel

**Comment by bluebill1049** [comment, marked as answer]
You can use only RHF, there is nothing stopping you from doing so, but I would recommend to use the right tool for the right job and keep `useForm` instance for per `<form />`  tag.

**Reply by awmiklovic** [reply]
Ah, I see. I overlooked that each step has its own useForm in the docs. What are the potential pitfalls of multiple `<form />` tags per useForm?

**Reply by bluebill1049** [reply]
have a  look at this: https://twitter.com/bluebill1049/status/1427213303898468361

**Reply by awmiklovic** [reply]
Thank you!
